sympy/assumptions/cnf.py

```diff
@@ -0,0 +1,373 @@
+from itertools import combinations, product
+
+from sympy import S
+from sympy.logic.boolalg import BooleanFunction, Or, And, Not
```

Comment:
It might be better to do `from sympy.logic import boolalg` and use `boolalg.Or`, and so on, to avoid confusion with the classes defined here.